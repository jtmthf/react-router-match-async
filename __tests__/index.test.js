import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router';
import MatchAsync from '../index';

function sleep(ms = 0) {
  return new Promise(r => setTimeout(r, ms));
}

async function load() {
  await sleep(200);
  return <p>Hello Async Route!</p>
}

it('loads components asynchronously', async () => {
  const component = renderer.create(
    <MemoryRouter>
      <MatchAsync exactly pattern="/" getComponent={load} />
    </MemoryRouter>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  await sleep(300);
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
