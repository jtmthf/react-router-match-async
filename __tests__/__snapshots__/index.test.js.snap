exports[`test loads components asynchronously 1`] = `null`;

exports[`test loads components asynchronously 2`] = `
<p>
  Hello Async Route!
</p>
`;
